hack/lib/version.sh

```diff
@@ -96,6 +96,13 @@ kube::version::get_version_vars() {
           KUBE_GIT_MINOR+="+"
         fi
       fi
+
+      # If KUBE_GIT_VERSION is not a valid Semantic Version, then refuse to build.
+      if ! [[ "${KUBE_GIT_VERSION}" =~ ^v([0-9]+)\.([0-9]+)(\.[0-9]+)?([-][0-9A-Za-z.-]+|[-][0-9A-Za-z.-]+[+][0-9A-Za-z.-]+|[+][0-9A-Za-z.-]+)?$ ]]; then
```

Comment:
minor nit: couldn't the last part of this be simplified to
`(-[0-9A-Za-z.-]+)?(\+[0-9A-Za-z.-])?`?

i.e. "an optional pre-release version followed by an optional build metadata section"?

Full regex would look something like `^v([0-9]+)\.([0-9]+)(\.[0-9]+)?(-[0-9A-Za-z.-]+)?(\+[0-9A-Za-z.-])?$`.

Reply:
Fixed.

Thanks, this regx is better.